terminal_solaris.go

```diff
@@ -1,15 +1,22 @@
+// Based on ssh/terminal/util_solaris.go:
+// Copyright 2015 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
 // +build solaris
 
 package logrus
 
 import (
-	"os"
-
 	"golang.org/x/sys/unix"
+	"syscall"
 )
 
 // IsTerminal returns true if the given file descriptor is a terminal.
+// see: http://src.illumos.org/source/xref/illumos-gate/usr/src/lib/libbc/libc/gen/common/isatty.c
 func IsTerminal() bool {
-	_, err := unix.IoctlGetTermios(int(os.Stdout.Fd()), unix.TCGETA)
+	fd := syscall.Stderr
+	var termio unix.Termio
+	err := unix.IoctlSetTermio(fd, unix.TCGETA, &termio)
```

Comment:
I'm confused as to why this is calling IoctlSetTermio instead of IoctlGetTermio, and I'm also confused as to why this is assuming stderr specifically instead of stdout.

Can you explain that a bit more?